src/rt/dmain2.d

```diff
@@ -468,8 +468,26 @@ extern (C) int _d_run_main(int argc, char **argv, MainFunc mainFunc)
     //       thrown during cleanup, however, will abort the cleanup process.
     void runAll()
     {
-        if (rt_init() && runModuleUnitTests())
-            tryExec({ result = mainFunc(args); });
+        if (rt_init())
+        {
+            int utResult = runModuleUnitTests();
+            if (utResult == 0)
+            {
+                tryExec({ result = mainFunc(args); });
+            }
+            else if (utResult > 0)
+            {
+                if(utResult != int.max)
```

Comment:
> Space after if

ok



> How about using the enum

The enum is in the public interface, not imported here. I could re-create the enum, but I don't think it's worth doing that. I will put a comment to clarify that it is attached to the enum.